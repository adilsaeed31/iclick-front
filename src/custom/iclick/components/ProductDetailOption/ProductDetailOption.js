import React, { Component } from "react";
import PropTypes from "prop-types";
import ButtonBase from "@material-ui/core/ButtonBase";
import Typography from "@material-ui/core/Typography";

export default class ProductDetailOption extends Component {
  static propTypes = {
    isActive: PropTypes.bool,
    onClick: PropTypes.func,
    option: PropTypes.object
  }

  handleOnClick = () => {
    this.props.onClick && this.props.onClick(this.props.option);
  }

  render() {
    const { option } = this.props;

    return (
      <ButtonBase disableRipple onClick={this.handleOnClick}>
        <Typography component="span" variant="body1">
          {option.optionTitle}
        </Typography>
      </ButtonBase>
    );
  }
}
