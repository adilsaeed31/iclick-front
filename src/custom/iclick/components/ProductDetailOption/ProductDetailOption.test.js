import React from "react";
import renderer from "react-test-renderer";
import { MuiThemeProvider } from "@material-ui/core/styles";
import { Provider } from "mobx-react";
import theme from "custom/reactionTheme";
import option from "./__mocks__/option.mock";
import ProductDetailOption from "./ProductDetailOption";

const uiStore = {
  pdpSelectedOptionId: "cmVhY3Rpb24vY2F0YWxvZ1Byb2R1Y3RWYXJpYW50OkpobjJ0YzVaRkFzYVdiaXZR"
};

test("basic snapshot", () => {
  const component = renderer.create((
    <MuiThemeProvider theme={theme}>
      <Provider uiStore={uiStore}>
        <ProductDetailOption
          onClick={() => true}
          isSelected
          option={option}
        />
      </Provider>
    </MuiThemeProvider>));
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
