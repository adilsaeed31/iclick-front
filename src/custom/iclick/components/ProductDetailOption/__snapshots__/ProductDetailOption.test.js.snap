// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic snapshot 1`] = `
<button
  className="MuiButtonBase-root-5 SkProductDetailOption-optionButton-1"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex="0"
  type="button"
>
  <span
    className="MuiTypography-root-8 MuiTypography-body1-17 SkProductDetailOption-optionText-2"
  >
    Blue
  </span>
</button>
`;
